src/core/namespace.c

```diff
@@ -2498,7 +2498,7 @@ int setup_namespace(
                 r = mount_nofollow_verbose(LOG_DEBUG, root, root, NULL, MS_BIND|MS_REC, NULL);
                 if (r < 0)
                         goto finish;
-                r = mount_switch_root(root, MOUNT_ATTR_PROPAGATION_INHERIT);
+                r = mount_switch_root(root, 0);
```

Comment:
You at least lost the nice clarity expressed by `MOUNT_ATTR_PROPAGATION_INHERIT` clearly stating that propagation is taken from the parent.